### ISSUE TITLE
FoodItemsController will not be garbage collected

### ISSUE THREAD

**Issue by firewave-remo** [FabianGosebrink/ASPNETCore-Angular-SignalR-Typescript#12]
I know this repo is just for demo purpose but i think you should correct this, before someone will use this in its own application.

You create a reference to the time elapsed event in the constructor of the FoodItemsController. This will prevent it from beeing garbage collected, because the TimeService has a Singleton lifestyle and will life forever.

```csharp
public FoodItemsController(IFoodRepository foodRepository, ITimerService timerService, IHubContext<CoolMessagesHub> hubContext)
        {
            _foodRepository = foodRepository;
            _coolMessageHubContext = hubContext;
            timerService.TimerElapsed += _timerService_TimerElapsed; // <-- creates a reference
        }
````

Second "bug":
In the event handler of the elapsed event you send a SignalR message to all connected clients, and since the controller will not be garbage collected you have thousands of messages sent around with the identical content to the clients at the same time after a short amount of time. 

But it would be iteresting so see where we should setup a background service like this one in asp net core.

**Comment by FabianGosebrink** [comment]
Hey @VSDekar , you are totally correct. As this repo is for demo purposes noone should copy this in his own template 1:1. So what would you recommend?

**Comment by thomasduft** [comment]
One idea could be to implement an [IHostedService](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.ihostedservice?view=aspnetcore-2.0) ([see](https://www.stevejgordon.co.uk/asp-net-core-2-ihostedservice)) together with some kind of an  application wide InMemoryEventSourceService (IMESE).

Then within the FoodItemsController you could just add events to the IMESE instead of attaching the _timerService_TimerElapsed handler.

The implementation of the IHostedService then needs to wake up and treats the hanging events in the queue accordingly. Most likely sending the events via the CoolMessageHubContext.

If time is left I will try to create a PR.

**Comment by FabianGosebrink** [comment]
closed via #14
